pkg/services/ngalert/api/util.go

```diff
@@ -111,6 +111,11 @@ func (p *AlertingProxy) withReq(
 	if err != nil {
 		return ErrResp(http.StatusBadRequest, err, "")
 	}
+	for h, vs := range ctx.Req.Header {
+		for _, v := range vs {
+			req.Header.Add(h, v)
+		}
+	}
 	for h, v := range headers {
```

Comment:
These headers are "extra" headers injected by grafana, not user headers. They are nil almost all the time, and rarely contain a Content-Type header

Reply:
Worth it to mention this in a comment, WDYT?